napari/_tests/test_magicgui.py

```diff
@@ -346,3 +346,21 @@ def test_layers_populate_immediately(make_napari_viewer):
     assert not len(labels_layer.choices)
     viewer.window.add_dock_widget(labels_layer)
     assert len(labels_layer.choices) == 1
+
+
+def test_from_layer_data_tuple(make_napari_viewer):
+    """Test that a function returning a layer data tuple runs without error."""
+    viewer = make_napari_viewer()
+
+    @magicgui
+    def from_layer_data_tuple() -> types.LayerDataTuple:
+        data = np.random.rand(10, 10)
+        meta = {'name': 'test_image'}
```

Comment:
Given that you are creating a `dict` here, I don't think this test fails on main.

I'd assume that the return values should be initialized with `Layer.as_layer_data_tuple` to check internal consistency within napari like in the reproducer in the related issue. That's how the bug arose in practice and it seems like a natural usage.